src/Http/CookieFactory.php

```diff
@@ -72,45 +71,34 @@ public function __construct(Config $config)
      * This method returns a cookie instance for use with the Set-Cookie HTTP header.
      * It will be pre-configured according to Flarum's base URL and protocol.
      *
-     * @param  string  $name
-     * @param  string  $value
-     * @param  int     $maxAge
-     * @return \Dflydev\FigCookies\SetCookie
+     * @param  string       $name
+     * @param  string|null  $value
+     * @param  int|null     $maxAge
+     * @return SetCookie
      */
-    public function make($name, $value = null, $maxAge = null)
+    public function make(string $name, ?string $value, ?int $maxAge): SetCookie
     {
-        $cookie = SetCookie::create($this->getName($name), $value);
-
-        // Make sure we send both the MaxAge and Expires parameters (the former
-        // is not supported by all browser versions)
-        if ($maxAge) {
-            $cookie = $cookie
-                ->withMaxAge($maxAge)
-                ->withExpires(time() + $maxAge);
-        }
-
-        if ($this->domain != null) {
-            $cookie = $cookie->withDomain($this->domain);
-        }
-
-        // Explicitly set SameSite value, use sensible default if no value provided
-        $cookie = $cookie->withSameSite(SameSite::{$this->samesite ?? 'lax'}());
-
-        return $cookie
-            ->withPath($this->path)
-            ->withSecure($this->secure)
-            ->withHttpOnly(true);
+        return new SetCookie(
+            $this->getName($name),
+            $value,
+            $maxAge ? time() + $maxAge : null,
+            $this->path,
+            $this->domain ?? '',
+            $this->secure,
+            true,
+            $this->samesite ?? 'lax'
+        );
     }
 
     /**
      * Make an expired cookie instance.
      *
      * @param string $name
-     * @return \Dflydev\FigCookies\SetCookie
+     * @return SetCookie
      */
-    public function expire($name)
+    public function expire(string $name): SetCookie
     {
-        return $this->make($name)->expire();
+        return SetCookie::thatExpires($name);
```

Comment:
Looks like we're not providing a value, but that doesn't appear to be a nullable field (https://github.com/hansott/psr7-cookies/blob/master/src/SetCookie.php#L36)? Do we have any automated tests that call this code?

Reply:
I don't know, I refactored existing code to be exactly as it was before.

Reply:
> It also doesn't make sense for a cookie to not have a value.

But this cookie right here doesn't have a value? Essentially I just want to make sure that doing it this way won't crash the site, and it seems like the new cookie library requires a non-nullable `$value` param

Reply:
Sure, I'll check it out.